### PR TITLE
Add `isCopy(Ty)` method

### PR DESCRIPTION
**Pull request by matklad**
r? @vlad20012

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/inspections/RsSelfConventionInspection.kt [hunk 1]
```diff
@@ -13,49 +13,53 @@ import org.rust.lang.core.psi.RsTraitItem
...
     NO_SELF("no self"),
-    SELF("self by value"),
-    REF_SELF("self by reference"),
-    REF_MUT_SELF("self by mutable reference");
+    BY_VAL("self by value"),
+    BY_REF("self by reference"),
+    BY_MUT_REF("self by mutable reference");
 }
 
 private val RsTraitOrImpl.traits: Sequence<RsTraitItem>
```
Comment (farodin91) [inline src/main/kotlin/org/rust/ide/inspections/RsSelfConventionInspection.kt]: You can remove this function and make supertraits private again.

Comment (matklad) [reply src/main/kotlin/org/rust/ide/inspections/RsSelfConventionInspection.kt]: Excellent, thanks @farodin91 !

File: src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt [hunk 2]
```diff
@@ -55,6 +56,9 @@ class ImplLookup(private val project: Project, private val items: StdKnownItems)
     val fnOutputParam by lazy(NONE) {
         RsLangItemIndex.findLangItem(project, "fn_once")?.let { findFreshAssociatedType(it, "Output") }
     }
+    private val copyTrait: RsTraitItem? by lazy(NONE) {
```
Comment (farodin91) [inline src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt]: Is it possbile to make this more generic?

Comment (matklad) [reply src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt]: Hm, could you elaborate a bit?

Comment (farodin91) [reply src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt]: I would change it to a function, than you say copy or debug, for example: driveTrait("Debug")...

Comment (matklad) [reply src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt]: Yeah, that makes sense, and we should definitely do it once we handle more then one derived trait specially. But for a single use case, I usually prefer not to create an abstraction upfront.

Comment (farodin91) [reply src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt]: Okay

**Pull request merged**